matkono/HealthSupApp
# Big refactoring from decision tree and login features

## What this PR does

- Refactor login feature;
- Refactor decision tree feature;
- Integrate the decision tree with API;
- Integrate functionality like get current node;
- Integrate functionality like get previous node;
- Integrate functionality like get next node;
- Check if the action was realized;
- New authentication flow to login and decision tree;
- New responsive framework to the UI;

## How to check if this PR is correct

- Try to login and check if login was successfully;
- Click 'Continue' in the patient details to get the current node of the decision tree;
- Click at the button 'Voltar' to get the previous node of the decision tree if it isn't the first node;
- Click at the button 'Avançar' to get the next node of the decision tree;
- If the decision tree falls into an action, a message appear warning that the action must be realized;

## Evidences

- First radio button question from API:
![image](https://user-images.githubusercontent.com/44846533/103143141-dd5e3980-46ef-11eb-8c17-30f7b3bdc266.png)

- Checkbox question from API:
![image](https://user-images.githubusercontent.com/44846533/103143153-f1a23680-46ef-11eb-961a-8f987c29c4ef.png)

- Decision node from API:
![image](https://user-images.githubusercontent.com/44846533/103143169-07176080-46f0-11eb-8eed-f72cd30d2ba4.png)

- Action node from API:
![image](https://user-images.githubusercontent.com/44846533/103143175-139bb900-46f0-11eb-90a6-a4435c26a9df.png)

- Verifying if action was realized:
![image](https://user-images.githubusercontent.com/44846533/103143180-24e4c580-46f0-11eb-88d9-0110787850a2.png)
